### PR TITLE
Tools/Build: Fix connection patcher for linux

### PR DESCRIPTION
**Pull request by StormBytePP**
This attempts to fix connection_patcher for linux.
Instead of giving an error std::wstring constructor ( see http://community.trinitycore.org/topic/11591-help-with-connection-patcher-under-linux/ ), now it creates Wow_patched file

### CODE REVIEW

**Comment by dansanti** [comment]
Nice with this patch, now i can run wow on ubuntu, thanks.

File: src/tools/connection_patcher/Program.cpp [hunk 1]
```diff
@@ -32,6 +32,9 @@
 
 #if PLATFORM == PLATFORM_WINDOWS
 #include <Shlobj.h>
+#elif PLATFORM == PLATFORM_UNIX
```
Comment (DDuarte) [inline src/tools/connection_patcher/Program.cpp]: Is this false for MacOS X?

Comment (StormBytePP) [inline src/tools/connection_patcher/Program.cpp]: I don't own a MAC to test, but what I can tell is that PLATFORM_LINUX is returning false in my machine (Linux)

Comment (DDuarte) [inline src/tools/connection_patcher/Program.cpp]: As I can see in the original code OS X was fully supported.

Answering my own question, yes it's false (PLATFORM_APPLE is used for OS X: https://github.com/TrinityCore/TrinityCore/blob/6.x/src/server/shared/CompilerDefs.h)

File: src/tools/connection_patcher/Program.cpp [hunk 2]
```diff
@@ -167,6 +173,18 @@ namespace Connection_Patcher
     }
 }
 
+#if PLATFORM == PLATFORM_UNIX
+std::wstring expand_user(std::string path)
+{
+    std::string expanded;
+    wordexp_t exp_result;
+    wordexp(std::string(path.begin(), path.end()).c_str(), &exp_result, 0);
+    expanded=exp_result.we_wordv[0];
```
Comment (DDuarte) [inline src/tools/connection_patcher/Program.cpp]: missing spaces between =

File: src/tools/connection_patcher/Program.cpp [hunk 3]
```diff
@@ -188,9 +206,11 @@ int main(int argc, char** argv)
         std::string const binary_path(std::move(vm["path"].as<std::string>()));
         std::string renamed_binary_path(binary_path);
 
-        wchar_t* commonAppData(nullptr);
 #if PLATFORM == PLATFORM_WINDOWS
+        wchar_t* commonAppData(nullptr);
         SHGetKnownFolderPath(FOLDERID_ProgramData, 0, NULL, &commonAppData);
+#else
+        std::wstring commonAppData=expand_user(vm["winProgramData"].as<std::string>());
```
Comment (DDuarte) [inline src/tools/connection_patcher/Program.cpp]: missing spaces between =

File: src/tools/connection_patcher/Program.cpp (hunk 1 above)
Comment (StormBytePP) [inline src/tools/connection_patcher/Program.cpp]: Nope: https://github.com/TrinityCore/TrinityCore/blob/6.x/src/tools/connection_patcher/Program.cpp#L192-L194

commonData variable (which is used as root path for adding bnet path to it) was only being initializated in Windows, so for linux and mac, a exception was being thrown due to it

Comment (DDuarte) [inline src/tools/connection_patcher/Program.cpp]: https://github.com/StormBytePP/TrinityCore/blob/6.x_fix_connection_patcher_linux/src/tools/connection_patcher/Program.cpp#L262-L283

**Pull request closed without merging**